src/types.d.ts

```diff
@@ -0,0 +1,2255 @@
+     * replaced the DDL target. Example case: the query is CREATE OR REPLACE
+     * TABLE, and the table already exists. &quot;DROP&quot;: The query deleted
+     * the DDL target.
+     */
+    ddlOperationPerformed?: string;
+    /**
+     * The DDL target table. Present only for CREATE/DROP TABLE/VIEW queries.
+     */
+    ddlTargetTable?: ITableReference;
+    /**
+     * [Output-only] The original estimate of bytes processed for the job.
+     */
+    estimatedBytesProcessed?: string;
+    /**
+     * [Output-only, Beta] Information about create model query job progress.
+     */
+    modelTraining?: IBigQueryModelTraining;
+    /**
+     * [Output-only, Beta] Deprecated; do not use.
+     */
+    modelTrainingCurrentIteration?: number;
+    /**
+     * [Output-only, Beta] Deprecated; do not use.
+     */
+    modelTrainingExpectedTotalIteration?: string;
+    /**
+     * [Output-only] The number of rows affected by a DML statement. Present
+     * only for DML statements INSERT, UPDATE or DELETE.
+     */
+    numDmlAffectedRows?: string;
+    /**
+     * [Output-only] Describes execution plan for the query.
+     */
+    queryPlan?: IExplainQueryStage[];
+    /**
+     * [Output-only] Referenced tables for the job. Queries that reference more
+     * than 50 tables will not have a complete list.
+     */
+    referencedTables?: ITableReference[];
+    /**
+     * [Output-only] Job resource usage breakdown by reservation.
+     */
+    reservationUsage?: Array<{name?: string; slotMs?: string;}>;
+    /**
+     * [Output-only] The schema of the results. Present only for successful dry
+     * run of non-legacy SQL queries.
+     */
+    schema?: ITableSchema;
+    /**
+     * The type of query statement, if valid. Possible values (new values might
+     * be added in the future): &quot;SELECT&quot;: SELECT query.
+     * &quot;INSERT&quot;: INSERT query; see
+     * https://cloud.google.com/bigquery/docs/reference/standard-sql/data-manipulation-language.
+     * &quot;UPDATE&quot;: UPDATE query; see
+     * https://cloud.google.com/bigquery/docs/reference/standard-sql/data-manipulation-language.
+     * &quot;DELETE&quot;: DELETE query; see
+     * https://cloud.google.com/bigquery/docs/reference/standard-sql/data-manipulation-language.
+     * &quot;MERGE&quot;: MERGE query; see
+     * https://cloud.google.com/bigquery/docs/reference/standard-sql/data-manipulation-language.
+     * &quot;CREATE_TABLE&quot;: CREATE [OR REPLACE] TABLE without AS SELECT.
+     * &quot;CREATE_TABLE_AS_SELECT&quot;: CREATE [OR REPLACE] TABLE ... AS
+     * SELECT ... . &quot;DROP_TABLE&quot;: DROP TABLE query.
+     * &quot;CREATE_VIEW&quot;: CREATE [OR REPLACE] VIEW ... AS SELECT ... .
+     * &quot;DROP_VIEW&quot;: DROP VIEW query. &quot;ALTER_TABLE&quot;: ALTER
+     * TABLE query. &quot;ALTER_VIEW&quot;: ALTER VIEW query.
+     */
+    statementType?: string;
+    /**
+     * [Output-only] [Beta] Describes a ti
```

Comment:
I agree with @JustinBeckwith, if this is just a copy of the definition in `googleapis` it would be nice to just use the definitions from the other library -- so that the two libraries don't drift apart.

I'd just create a helper in the `scripts/` folder that copies this file over using `require.resolve` (if I'm understanding the problem correctly).

Reply:
While I definitely agree if we could just borrow the types, that is the best options. But the apiary types file pulls in dependencies we don't use and contain a number of fields we don't support. 

@JustinBeckwith how are the types generated for apiary? The naming convention makes me think they were generated via the discovery api. Do we see a benefit in writing a cli tool (or something) that generates types based on those responses?

Reply:
They're all generated by the apiary generator:
https://github.com/googleapis/google-api-nodejs-client/blob/master/src/generator/generator.ts

To be honest - I don't really mind if you copy the types out :) Whatever makes sense.

Reply:
Oh nice, I had actually started hacking away at something to create the types. I noticed that the apiary types were missing a few thing

* doesn't look to support `any` types, looks like an empty interface is generated instead (See [JsonValue](https://github.com/googleapis/google-api-nodejs-client/blob/master/src/apis/bigquery/v2.ts#L1631))
* doesn't look to support `additionalProperties` field, also generates an empty interface (See [JsonObject](https://github.com/googleapis/google-api-nodejs-client/blob/master/src/apis/bigquery/v2.ts#L1630)) but should be something like
   ```typescript
    interface JsonObject {
      [key: string]: JsonValue;
    }
   ```

Reply:
So just a heads up, I wrote a little CLI tool to generate these via Discovery API. I think there's a couple more things to do before it can be open sourced, but once that happens we can add it as a dev dep and just generate them easily whenever. In the meantime I've pushed another commit that includes the generated types!